src/main/java/sirius/biz/storage/layer2/BlobSoftRefProperty.java

```diff
@@ -64,6 +87,120 @@ private BlobSoftRefProperty(@Nonnull EntityDescriptor descriptor,
         super(descriptor, accessPath, field);
     }
 
+    /**
+     * Returns the {@link EntityDescriptor} of the referenced entity.
+     *
+     * @return the referenced entity descriptor
+     */
+    public List<EntityDescriptor> getReferencedDescriptors() {
+        if (referencedDescriptors == null) {
+            if (blobSoftRef == null) {
+                throw new IllegalStateException("Schema not linked!");
+            }
+            referencedDescriptors.add(mixing.getDescriptor(MongoBlob.class));
+            referencedDescriptors.add(mixing.getDescriptor(SQLBlob.class));
+        }
+
+        return Collections.unmodifiableList(referencedDescriptors);
+    }
+
+    private BlobSoftRef getBlobSoftRef() {
+        if (blobSoftRef == null) {
+            try {
+                this.blobSoftRef = (BlobSoftRef) field.get(accessPath.apply(descriptor.getReferenceInstance()));
+            } catch (Exception e) {
+                throw Exceptions.handle()
+                                .to(Mixing.LOG)
+                                .error(e)
+                                .withSystemErrorMessage(
+                                        "Unable to obtain a BlobSoftRef object from entity ref field ('%s' in '%s'): %s (%s)",
+                                        getName(),
+                                        descriptor.getType().getName())
+                                .handle();
+            }
+        }
+
+        return blobSoftRef;
+    }
+
+    @Override
+    protected void link() {
+        super.link();
+
+        try {
+            BaseEntityRef.OnDelete deleteHandler = getBlobSoftRef().getDeleteHandler();
+
+            if (deleteHandler == BaseEntityRef.OnDelete.CASCADE) {
+                getReferencedDescriptors().forEach(entityDescriptor -> entityDescriptor.addCascadeDeleteHandler(this::onDeleteCascade));
```

Comment:
forEachBlobType(entityDescriptor -> entityDescriptor.addCascadeDeleteHandler(this::onDeleteCascade));